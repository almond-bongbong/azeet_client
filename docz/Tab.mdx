---
name: Tab
---

import { Playground, Props } from 'docz'
import Tab from 'components/Tab/Tab'

# Tab

<Props of={Tab} />

## Basic usage

<Playground>
  {() => {
    const Test = () => {
      const [value, setValue] = React.useState('jp');

      return (
        <Tab
          value={value}
          onClick={setValue}
          items={[
            { value: "ko", title: "한국" },
            { value: "jp", title: "일본" },
            { value: "hk", title: "홍콩" },
            { value: "mo", title: "마카오" },
          ]}
        />
      );
    }

    return (
      <Test />
    )
  }}
</Playground>
