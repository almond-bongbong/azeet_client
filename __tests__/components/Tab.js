import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Tab from 'components/Tab/Tab';

describe('Select', () => {
  it('matches snapshot', () => {
    const utils = render(
      <Tab
        onClick={_ => _}
        value="ko"
        items={[
          { value: 'ko', title: '한국' },
          { value: 'jp', title: '일본' },
          { value: 'hk', title: '홍콩' },
          { value: 'mo', title: '마카오' },
        ]}
      />,
    );

    expect(utils.container).toMatchSnapshot();
  });

  it('selected second tab', () => {
    const { getByText } = render(
      <Tab
        onClick={_ => _}
        value="jp"
        items={[
          { value: 'ko', title: '한국' },
          { value: 'jp', title: '일본' },
          { value: 'hk', title: '홍콩' },
          { value: 'mo', title: '마카오' },
        ]}
      />,
    );

    expect(getByText('일본').closest('button')).toHaveStyle('color: #1890ff');
    expect(getByText('한국').closest('button')).not.toHaveStyle('color: #1890ff');
  });

  it('calls onClick hk', () => {
    const handleClick = jest.fn();
    const { getByText } = render(
      <Tab
        onClick={handleClick}
        value="jp"
        items={[
          { value: 'ko', title: '한국' },
          { value: 'jp', title: '일본' },
          { value: 'hk', title: '홍콩' },
          { value: 'mo', title: '마카오' },
        ]}
      />,
    );

    const hk = getByText('홍콩');
    fireEvent.click(hk);
    expect(handleClick).toBeCalledWith('hk');
  });
});
