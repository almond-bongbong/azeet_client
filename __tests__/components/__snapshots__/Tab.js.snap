// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select matches snapshot 1`] = `
<div>
  <div
    class="Tab__TabStyle-pfmx17-0 iKWPxz"
  >
    <div
      class="tab-nav"
    >
      <button
        class="nav-item active"
        type="button"
      >
        <span>
          한국
        </span>
      </button>
      <button
        class="nav-item "
        type="button"
      >
        <span>
          일본
        </span>
      </button>
      <button
        class="nav-item "
        type="button"
      >
        <span>
          홍콩
        </span>
      </button>
      <button
        class="nav-item "
        type="button"
      >
        <span>
          마카오
        </span>
      </button>
    </div>
    <div
      class="Tab__BarStyle-pfmx17-1 fFOGPD"
      width="0"
    />
  </div>
</div>
`;
